### PR TITLE
Verify profile with external applications

### LINKED ISSUE #472: Verify profile with centralized applications
## Feature description 
All the discussion has taken place inside #364

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description
This PR implements the feature that allows users to connect their Desmos profile with centralized applications. It represents the result of the work done inside the following PRs: 
- #473
- #479 
- #484

It closes #472

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit tests.
- [x] Wrote integration tests (simulation & CLI). 
- [x] Updated the documentation. 
- [x] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: docs/developers/msgs/profiles/link-application.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# `MsgLinkApplication`
+This message allows you start the process that will verify
```
Comment (bragaz) [inline docs/developers/msgs/profiles/link-application.md:2]:
```suggestion
This message allows you to start the process that will verify
```

File: docs/developers/msgs/profiles/unlink-application.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# `MsgUnlinkApplication`
+This message allows you remove a previously linked application from your Desmos profile.
```
Comment (bragaz) [inline docs/developers/msgs/profiles/unlink-application.md:2]:
```suggestion
This message allows you to remove a previously linked application from your Desmos profile.
```

File: docs/developers/msgs/profiles/unlink-application.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# `MsgUnlinkApplication`
+This message allows you to remove a previously linked application from your Desmos profile.
+
+## Structure
+
+```json
+{
+  "@type": "/desmos.profiles.v1beta1.MsgUnlinkApplication",
+  "application": "<Name of the application to be unlinked>",
```
Comment (bragaz) [inline docs/developers/msgs/profiles/unlink-application.md:9]:
```suggestion
  "application": "<Name of the application to unlink>",
```

File: docs/developers/msgs/profiles/unlink-application.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "username": "<Name of the account inside the application that should be unlinked>",
+  "signer": "<Desmos address of the profile that should remove the link>"
+}
+```
+
+### Attributes
+
+| Attribute | Type | Description |
+| :-------: | :----: | :-------- |
+| `application`  | `string` | Name of the application to be unlinked |
```
Comment (bragaz) [inline docs/developers/msgs/profiles/unlink-application.md:19]:
```suggestion
| `application`  | `string` | Name of the application to unlink |
```

File: docs/developers/types/profiles/application-link.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# Application link
+An application link (abbr. _app link_) represents a link to an external (and possibly centralized) application. These links are created by the user and their validity is checked using a multi-step verification process described inside the [_"Themis"_ repository](https://github.com/desmos-labs/themis). 
+
+## `User`
+The `User` represents the address of the Desmos profile to which the link is associated.
```
Comment (bragaz) [inline docs/developers/types/profiles/application-link.md:5]:
```suggestion
Address of the Desmos profile to which the link is associated. 
```

File: docs/developers/types/profiles/application-link.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+# Application link
+An application link (abbr. _app link_) represents a link to an external (and possibly centralized) application. These links are created by the user and their validity is checked using a multi-step verification process described inside the [_"Themis"_ repository](https://github.com/desmos-labs/themis). 
+
+## `User`
+The `User` represents the address of the Desmos profile to which the link is associated. 
+
+## `Data`
+The `Data` object contains the details of the link.
```
Comment (bragaz) [inline docs/developers/types/profiles/application-link.md:8]:
```suggestion
Object that contains the details of the link. 
```

File: docs/developers/types/profiles/application-link.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+The `Data` object contains the details of the link. 
+
+### `Application`
+Name of the application to which the link refers to (eg. `twitter`, `github`, `reddit`, etc). 
+
+### `Username`
+Identifier of the application account which the link refers to (eg. Twitter username, GitHub profile, Reddit username, etc).
+
+## `State`
+The `State` field represents the current state of the link. There can be five different states in which a link can be:
```
Comment (bragaz) [inline docs/developers/types/profiles/application-link.md:17]:
```suggestion
Representation of the current state of the link. There can be five different states in which a link can be: 
```

File: docs/developers/types/profiles/application-link.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+### `OracleScriptID`
+Unique id of the script that has been called to verify the authenticity of the link. 
+
+### `CallData`
+Contains the details of the data that will be used to call the oracle script. 
+
+#### `Application`
+Name of the application for which the link is valid (eg. `twitter`, `github`, `reddit`, etc). 
+
+#### `CallData`
```
Comment (bragaz) [inline docs/developers/types/profiles/application-link.md:40]: repeated above

File: docs/developers/types/profiles/application-link.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+Represents a successful result. It contains two fields. 
+
+#### `Value`
+Plain text value that has been signed from the user with their Desmos private key to prove the ownership of the Desmos profile. 
+
+#### `Signature`
+Hex-encoded result of the plain text value signature. 
+
+### `Result_Error`
+Used to identify an error during the verification process. It contains only one field.
```
Comment (bragaz) [inline docs/developers/types/profiles/application-link.md:63]:
```suggestion
Identifies an error during the verification process. It contains only one field.
```

File: docs/developers/types/profiles/application-link.md (hunk 8 above)
Comment (RiccardoM) [reply docs/developers/types/profiles/application-link.md:40]: These are actually two fields: 
```json
{
  "oracle_request": {
    "call_data": {
      "application": "...",
      "call_data": "..."
    }
  }
}
```

**Review by bragaz — COMMENTED** [review]
Some small grammar stuff in docs

**Review by bragaz — APPROVED** [review]

**Review by bragaz — APPROVED** [review]

**Pull request merged**